exporter/keys.go

```diff
@@ -188,38 +139,54 @@ func (e *Exporter) extractCountKeysMetrics(ch chan<- prometheus.Metric, c redis.
 	}
 }
 
-func scanForKeyCount(c redis.Conn, pattern string) (int, error) {
-	iter := 0
-	keys := 0
+func getKeysCount(c redis.Conn, pattern string, count int64) (int, error) {
+	keysCount := 0
 
-	for {
-		arr, err := redis.Values(doRedisCmd(c, "SCAN", iter, "MATCH", pattern))
-		if err != nil {
-			return keys, fmt.Errorf("error retrieving '%s' keys err: %s", pattern, err)
-		}
-		if len(arr) != 2 {
-			return keys, fmt.Errorf("invalid response from SCAN for pattern: %s", pattern)
-		}
+	keys, err := scanKeys(c, pattern, count)
+	if err != nil {
+		return keysCount, fmt.Errorf("error retrieving '%s' keys err: %s", pattern, err)
+	}
+	keysCount = len(keys)
 
-		k, _ := redis.Values(arr[1], nil)
-		keys += len(k)
+	return keysCount, nil
+}
 
-		if iter, _ = redis.Int(arr[0], nil); iter == 0 {
-			break
+// getKeysFromPatterns does a SCAN for a key if the key contains pattern characters
+func getKeysFromPatterns(c redis.Conn, keys []dbKeyPair, count int64) (expandedKeys []dbKeyPair, err error) {
+	expandedKeys = []dbKeyPair{}
+	for _, k := range keys {
+		if regexp.MustCompile(`[\?\*\[\]\^]+`).MatchString(k.key) {
```

Comment:
I know this is existing code but can you pull `regexp.MustCompile(`[\?\*\[\]\^]+`)` out of the loop and instantiate it once as a const/var right outside this function and then just call it here?